tools/release/lib/release.py

```diff
@@ -1,12 +1,20 @@
+#!/usr/bin/env python3
+
 import re
+from typing import Optional, List, Set, Union, Type
+from github.PullRequest import PullRequest
+from github.GitRelease import GitRelease
+from github.Repository import Repository
 
 
-def is_valid_tag(tag):
+def is_valid_tag(tag: str) -> bool:
```

Comment:
Can you add a `ruff` rule to force using types in Python?

Reply:
Sure thing!

Reply:
Just a small consideration, is this something we really want? I'm sorry I added this (type hints) in the code because it was growing and I just didn't want to be the only person understanding this. I wanted to make the code more readable and reduce the time required for new developers to make changes to it. I mean, I really like python with type hints and personally I would be very happy to have it as a rule, but I don't know, maybe someone doesn't feel that way. cc @lemire

Reply:
Anyway, I added "ANN" (#flake8-annotations) in the mean time. Works like a charm!